.gitignore

```diff
@@ -30,3 +30,4 @@ docs/VERSION
 docs/GITCOMMIT
 docs/changed-files
 autogen/
+Dockerfile
```

Comment:
No matter what solution we end up with, IMO we'll still need a `Dockerfile` checked into the repo itself (even if it's generated from all the default options of our generation method).